resources/ruby.rb

```diff
@@ -57,9 +63,21 @@
     environment new_resource.environment if new_resource.environment
     action :run
     live_stream true if new_resource.verbose
-  end unless ruby_installed?
+    not_if { ruby_installed? && new_resource.rbenv_action != 'uninstall' }
+  end
+
+  log_message = new_resource.to_s
+  log_message << if new_resource.rbenv_action == 'uninstall'
+                   ' uninstalled'
+                 else
+                   " build time was #{(Time.now - install_start) / 60.0} minutes"
+                 end
+  Chef::Log.info(log_message)
```

Comment:
Tidy me up please 👍

Reply:
Yeah, I'm not happy with this construct either. How would you like me to resolve this?

Reply:
I take it back, the logging here is nice and only triggered on the resource firing. The log resource often just fires every time.